store/create-cafs-store/src/index.ts

```diff
@@ -34,7 +35,17 @@ export function createPackageImporterAsync (
   const gfm = getFlatMap.bind(null, opts.cafsDir)
   return async (to, opts) => {
     const { filesMap, isBuilt } = gfm(opts.filesResponse, opts.sideEffectsCacheKey)
-    const pkgImportMethod = (opts.requiresBuild && !isBuilt)
+    let requiresBuild = opts.requiresBuild
+    if (requiresBuild == null) {
```

Comment:
```suggestion
    if (requiresBuild == true) {
```

I think you meant requiresBuild == true?

Reply:
If you check my other changes in the PR, I want requiresBuild to be passed in as `undefined` if the package is optional.

Reply:
I see that now, how about if it got passed in like this? I don't love the nested ternaries but this would mean that if the item doesn't have a requiresBuild in the lockfile, it doesn't pass through the check.

This means that if we find a way to make the flag more accurate in the lock file (either through my PR  or in future, it could be incrementally adopted)

```suggestion
  requiresBuild: depNode.patchFile != null || (depNode.optional ? (depNode.requiresBuild === true ? undefined : false) : depNode.requiresBuild)
```

Reply:
Yeah, I think I can do this.